packages/flame/lib/src/collisions/raycast_result.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flame/collisions.dart';
+import 'package:flame/extensions.dart';
+import 'package:flame/src/geometry/ray2.dart';
+
+/// The result of a raycasting operation.
+///
+/// Note that the members of this class is heavily re-used. If you want to
+/// keep the result in an object, clone the parts you want, or the whole
+/// [RaycastResult] with [clone].
+class RaycastResult<T extends Hitbox<T>> {
+  RaycastResult({
+    T? hitbox,
+    Ray2? reflectionRay,
+    Vector2? normal,
+    double? distance,
+    this.isWithin = false,
+    this.isActive = true,
+  }) {
+    _hitbox = hitbox;
+    _reflectionRay = reflectionRay ?? Ray2.zero();
+    _normal = normal ?? Vector2.zero();
+    _distance = distance ?? double.maxFinite;
+  }
+
+  /// Whether this result has active results in it.
+  ///
+  /// This is used so that the objects in there can continue to live even when
+  /// there is no result from a ray cast.
+  bool isActive;
+
+  /// Whether the origin of the ray was within the hitbox.
+  bool isWithin;
```

Comment:
> Perhaps call it isInsideHitbox or something like that, for better clarity.

Good idea, I'll change the name.

>  Also it might be good to make this private (with getter), so that the user wouldn't be able to overwrite it.

With an internal setter? Because these objects needs to be possible to re-use.

Reply:
Changed them to private and only made them changable from `setWith`.